runtime/lib/ttnn/runtime.cpp

```diff
@@ -166,34 +438,112 @@ void deallocateBuffers(Device deviceHandle) {
+  const ::ttnn::Tensor &hostTensor =
+      tensor.as<::ttnn::Tensor>(DeviceRuntime::TTNN);
+  LOG_ASSERT(utils::isOnHost(hostTensor.storage_type()),
+             "Input tensor to ToDevice must be on host");
+  const ::ttnn::Layout &inputLayout = hostTensor.get_layout();
+  const ::ttnn::DataType &inputDataType = hostTensor.get_dtype();
+  LayoutDesc inputLayoutDesc(::ttnn::BufferType::SYSTEM_MEMORY, inputLayout,
+                             inputDataType, std::nullopt);
+  const LayoutDesc &outputLayoutDesc =
+      layout.as<LayoutDesc>(DeviceRuntime::TTNN);
+  LOG_ASSERT(outputLayoutDesc.isOnDevice(), "Output layout must be on device");
+  LOG_ASSERT(outputLayoutDesc.memoryConfig.has_value(),
+             "Output layout must have memory config");
   ::ttnn::MeshDevice &meshDevice =
-      deviceHandle.as<::ttnn::MeshDevice>(DeviceRuntime::TTNN);
-  std::vector<::ttnn::Tensor *> inputs;
-  inputs.reserve(inputHandles.size());
-  for (auto &input : inputHandles) {
-    LOG_ASSERT(input.matchesRuntime(DeviceRuntime::TTNN));
-    inputs.push_back(static_cast<::ttnn::Tensor *>(input.handle.get()));
-  }
+      device.as<::ttnn::MeshDevice>(DeviceRuntime::TTNN);
+  DeviceVariant targetDevice = getTargetDevice(meshDevice);
 
-  std::vector<::ttnn::Tensor *> outputs;
-  outputs.reserve(outputHandles.size());
-  for (auto &output : outputHandles) {
-    LOG_ASSERT(output.matchesRuntime(DeviceRuntime::TTNN));
-    outputs.push_back(static_cast<::ttnn::Tensor *>(output.handle.get()));
-  }
+  detail::LayoutConverter converter(inputLayoutDesc, outputLayoutDesc);
+  std::shared_ptr<::ttnn::Tensor> out = std::make_shared<::ttnn::Tensor>(
+      converter.convertTensorLayout(hostTensor, targetDevice));
 
-  tt::runtime::ttnn::runProgram(meshDevice, executableHandle, programIndex,
-                                inputs, outputs);
-  return Event(nullptr, DeviceRuntime::TTNN);
+  return Tensor(std::static_pointer_cast<void>(out), nullptr,
+                DeviceRuntime::TTNN);
 }
 
-void wait(Event event) {
-  // Not implemented
-  LOG_ASSERT(event.matchesRuntime(DeviceRuntime::TTNN));
+Tensor toLayout(Tensor tensor, Layout layout) {
+  const ::ttnn::Tensor &inputTensor =
+      tensor.as<::ttnn::Tensor>(DeviceRuntime::TTNN);
+  const ::ttnn::StorageType &inputStorageType = inputTensor.storage_type();
+  const ::ttnn::BufferType &inputBufferType =
+      utils::isOnHost(inputStorageType)
+          ? ::ttnn::BufferType::SYSTEM_MEMORY
+          : inputTensor.memory_config().buffer_type;
+  const ::ttnn::Layout &inputLayout = inputTensor.get_layout();
+  const ::ttnn::DataType &inputDataType = inputTensor.get_dtype();
+  std::optional<::ttnn::MemoryConfig> inputMemoryConfig = std::nullopt;
+  if (utils::isOnDevice(inputStorageType)) {
+    inputMemoryConfig =
+        std::make_optional<::ttnn::MemoryConfig>(inputTensor.memory_config());
+  }
+  LayoutDesc inputLayoutDesc(inputBufferType, inputLayout, inputDataType,
```

Comment:
Is it just me, or would this be readable with some whitespace (new lines) in meaningful places, after scopes, group initialization of variables/consts, etc. 😄

Reply:
Yeah I was writing this at midnight everything was dark so wanted to keep it consistent to the environment therefore no whitespaces :)))). Jokes aside I'll add whitespaces to this and format it